sklearn/compose/tests/test_target.py

```diff
@@ -37,7 +37,7 @@ def test_transform_target_regressor_error():
         regr.fit(X, y)
     # fit with sample_weight with a regressor which does not support it
     sample_weight = np.ones((y.shape[0],))
-    regr = TransformedTargetRegressor(regressor=Lasso(),
+    regr = TransformedTargetRegressor(regressor=OrthogonalMatchingPursuit(),
```

Comment:
Why this change?

Reply:
Here, one needs an estimator that does not support `sample_weight`. As `Lasso` inherits sample_weight support from `ElasticNet`, I needed to change this.